DashSync/shared/Models/Wallet/DSAccount.m

```diff
@@ -895,21 +879,22 @@ __block __weak BOOL (^_isAscending)(id, id) = isAscending = ^BOOL(DSTransaction
             }] != NSNotFound) return NO;
             if ([self.invalidTransactionHashes containsObject:hash1] && ![self.invalidTransactionHashes containsObject:hash2]) return YES;
             if ([self.pendingTransactionHashes containsObject:hash1] && ![self.pendingTransactionHashes containsObject:hash2]) return YES;
-            for (DSTransactionInput *input in tx1.inputs) {
-                if (_isAscending(self.allTx[uint256_obj(input.inputHash)], tx2)) return YES;
-            }
+
             return NO;
         };
         
+        NSArray *externalAddresses = self.externalAddresses;
+        NSArray *internalAddresses = self.internalAddresses;
```

Comment:
`self.externalAddresses` and `self.internalAddresses` was calculated from scratch on every tx comparison